lingvo/tasks/asr/ctc_model.py

```diff
@@ -138,62 +140,118 @@ def __init__(self, params):
 
   def ComputePredictions(self, theta, input_batch):
     output_batch = self._FProp(theta, input_batch)
-    return py_utils.RunOnTpuHost(
+    # ctc_greedy_decoder = merge_repeated = True (def)
+    hypotheses = py_utils.RunOnTpuHost(
       tf.nn.ctc_greedy_decoder,
       output_batch.encoder_outputs,
       py_utils.LengthsFromBitMask(
-        tf.squeeze(output_batch.encoder_outputs_padding, 2), 0)
+        tf.squeeze(output_batch.encoder_outputs_padding, 2), 0),
     )
+    return hypotheses
+        # return tf.Print(hypotheses, [hypotheses], "*******ComputePredictions********:")
+
+  def _CalculateWER(self, input_batch, output_batch):
+      (decoded,), neg_sum_logits = tf.nn.ctc_greedy_decoder(
+          output_batch.encoder_outputs,
+          py_utils.LengthsFromBitMask(
+              tf.squeeze(output_batch.encoder_outputs_padding, 2), 0
+          ),
+      )
+
+      dec = tf.sparse_to_dense(
+          decoded.indices, decoded.dense_shape, decoded.values, default_value=73
```

Comment:
No need to  use 73 as the default value here. We just need to get the padding right, and the default values can be arbitrary.